### PR TITLE
Fix jstdPhantom test errors - #367

### PR DESCRIPTION
**Pull request by chuckhays**
Fixes issue #367.

The environment in which the jstd tests run is different from the full
web browser. Anything that uses html elements directly and is executed
when a js file is loaded needs to be null checked.

@jiayliu

### CODE REVIEW

File: samples/web/content/apprtc/js/appwindow.js [hunk 1]
```diff
@@ -61,10 +61,12 @@ var params = {
...
 params.roomId = randomString(9);
 params.roomLink =  'https://apprtc.appspot.com/room/' + params.roomId;
-params.roomServer = 'https://apprtc.appspot.com';
+params.server = 'https://apprtc.appspot.com';
 
-var joinRoomLink = document.querySelector('#room-link');
-joinRoomLink.href = params.roomLink;
-joinRoomLink.text = params.roomLink;
+var joinRoomLink = document.querySelector('#room-link-href');
+if (joinRoomLink) {
```
Comment (jiayliu) [inline samples/web/content/apprtc/js/appwindow.js]: I think it's better to inject any necessary DOM elements in the test, than adding checks in the prod code.

File: samples/web/content/apprtc/js/appwindow.js [hunk 2]
```diff
@@ -61,10 +61,12 @@ var params = {
 // Generate random room id and connect.
 params.roomId = randomString(9);
 params.roomLink =  'https://apprtc.appspot.com/room/' + params.roomId;
-params.roomServer = 'https://apprtc.appspot.com';
+params.server = 'https://apprtc.appspot.com';
```
Comment (jiayliu) [inline samples/web/content/apprtc/js/appwindow.js]: 'roomServer' seems clearer to me.

File: samples/web/content/apprtc/js/appwindow.js [hunk 3]
```diff
@@ -61,10 +61,12 @@ var params = {
...
 
-var joinRoomLink = document.querySelector('#room-link');
-joinRoomLink.href = params.roomLink;
-joinRoomLink.text = params.roomLink;
+var joinRoomLink = document.querySelector('#room-link-href');
```
Comment (jiayliu) [inline samples/web/content/apprtc/js/appwindow.js]: Move the constant to UI_CONSTANTS? I'd like to keep all HTML constants in one place.

File: samples/web/content/apprtc/js/appwindow.js (hunk 2 above)
Comment (chuckhays) [inline samples/web/content/apprtc/js/appwindow.js]: Had to line up with this change from the refactoring:
+var Call = function(params) {
+  this.params_ = params;
+  this.roomServer_ = params.server || '';

I could change it to roomServer in the other place if you prefer?

File: samples/web/content/apprtc/js/appwindow.js (hunk 3 above)
Comment (chuckhays) [inline samples/web/content/apprtc/js/appwindow.js]: I did this in another CL, this was just the minimum required to get the Chrome App working again after the refactor. I can either do it here, or leave it in the other CL. Preference?

File: samples/web/content/apprtc/js/appwindow.js (hunk 2 above)
Comment (jiayliu) [inline samples/web/content/apprtc/js/appwindow.js]: Oops, that was a mistake of merging based on your old commit.


On Wed, Jan 14, 2015 at 10:01 AM, Chuck Hays <notifications@github.com>
wrote:

> In samples/web/content/apprtc/js/appwindow.js
> <https://github.com/GoogleChrome/webrtc/pull/374#discussion-diff-22955566>
> :
>
> > @@ -61,10 +61,12 @@ var params = {
> >  // Generate random room id and connect.
> >  params.roomId = randomString(9);
> >  params.roomLink =  'https://apprtc.appspot.com/room/' + params.roomId;
> > -params.roomServer = 'https://apprtc.appspot.com';
> > +params.server = 'https://apprtc.appspot.com';
>
> Had to line up with this change from the refactoring:
> +var Call = function(params) {
>
>    - this.params_ = params;
>    - this.roomServer_ = params.server || '';
>
> I could change it to roomServer in the other place if you prefer?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleChrome/webrtc/pull/374/files#r22955566>.
>



-- 
Jiayang Liu

File: samples/web/content/apprtc/js/appwindow.js (hunk 1 above)
Comment (chuckhays) [inline samples/web/content/apprtc/js/appwindow.js]: I'll investigate how to do this.

File: samples/web/content/apprtc/js/appwindow.js (hunk 3 above)
Comment (jiayliu) [inline samples/web/content/apprtc/js/appwindow.js]: I'm fine with doing it in the next CL

File: samples/web/content/apprtc/js/appwindow.js (hunk 1 above)
Comment (chuckhays) [inline samples/web/content/apprtc/js/appwindow.js]: I had made this change before I excluded appwindow.js, so it isn't actually needed right now, I reverted it. I haven't discovered how to manipulate the DOM yet with our test set up. The ways suggested for phantomJS don't work.

**Comment by chuckhays** [comment]
Updated.

File: samples/web/content/apprtc/js/util.js [hunk 4]
```diff
@@ -85,14 +85,15 @@ function filterTurnUrls(urls, protocol) {
...
 // Start shims for fullscreen
-
-document.cancelFullScreen = document.webkitCancelFullScreen ||
-document.mozCancelFullScreen || document.cancelFullScreen;
-
-document.body.requestFullScreen = document.body.webkitRequestFullScreen ||
-document.body.mozRequestFullScreen || document.body.requestFullScreen;
-
-document.onfullscreenchange = document.onwebkitfullscreenchange = document.onmozfullscreenchange;
+function setUpFullScreen() {
```
Comment (jiayliu) [inline samples/web/content/apprtc/js/util.js]: Add this to "exported" of this file, and "globals" of appcontroller.js

Comment (chuckhays) [inline samples/web/content/apprtc/js/util.js]: Fixed this, and a couple others. I was too focused on the jstd tests.

**Pull request merged**